src/gp_instance.cpp

```diff
@@ -37,11 +58,34 @@ void GPInstance::MakeEngine() {
   CheckSequencesAndTreesLoaded();
   ProcessLoadedTrees();
   SitePattern site_pattern(alignment_, tree_collection_.TagTaxonMap());
-  engine_ = std::make_unique<GPEngine>(
-      // To count GPCSSs, we add the usual suspects to the number of leaves (which are
-      // the fake PCSS).
-      site_pattern, sbn_parameters_.size() + tree_collection_.TaxonCount(),
-      mmap_file_path_);
+
+  ConstructDAG();
```

Comment:
In thinking about this before I had thought that there would be a separate GPDAG class, defined in gp_dag.* files. If it would only take an hour or two, I think that would be cleaner. What do you think?